repose-aggregator/components/filters/ip-user-filter/src/test/scala/org/openrepose/filters/ipuser/IpUserFilterTest.scala

```diff
@@ -159,6 +159,17 @@ class IpUserFilterTest extends FunSpec with BeforeAndAfterEach with Matchers {
     }
   }
 
+  describe("when presented with non-sense") {
+    it("should return a 400 when X-Forwarded-For is malformed") {
+      servletRequest.setRemoteAddr("10.1.2.3")
+      servletRequest.addHeader("X-Forwarded-For", "banana-phone")
+
+      ipUserFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      servletResponse.getStatus shouldBe 400
+    }
```

Comment:
Perhaps add a test to cover if `setRemoteAddr` returns a bad value? That could happen if a bad wrapper is used.

Reply:
Since tests are not exposed to the user in any way, I think a test for this is still valuable. It shows that we considered the case in question and this is the behavior that was decided on.